src/App.tsx

```diff
@@ -1,18 +1,59 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
+const updateTitle = (currentPage: number, countPerPage: number) => {
+  const isLastPage = ((currentPage - 1) * countPerPage + 1)
+  === items.length - 1;
+
+  const from = (currentPage - 1) * countPerPage + 1;
+  const to = isLastPage
+    ? items.length
+    : ((currentPage - 1) * countPerPage) + countPerPage;
+
+  return `Page ${currentPage} (items ${from} - ${to} of ${items.length})`;
+};
+
+const getVisibleItems = (currentPage: number, countPerPage: number) => {
+  const start = (currentPage - 1) * countPerPage;
+  const end = countPerPage + start;
+
+  return items.slice(start, end);
+};
+
 export const App: React.FC = () => {
+  const [countPerPage, setCountPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [visibleItems, setVisibleItems] = useState(items
+    .slice(0, countPerPage));
+
+  useEffect(() => {
+    const newVisibleItems = getVisibleItems(currentPage, countPerPage);
+
+    setVisibleItems(newVisibleItems);
+  }, [countPerPage, currentPage]);
+
+  const changeCountPerPage = (newCount: number) => {
+    setCountPerPage(newCount);
+    setCurrentPage(1);
+  };
+
+  const changeCurrentPage = (current: number) => {
+    setCurrentPage(current);
+  };
```

Comment:
we now create theses methods on each render. Would be better to wrap them into a `useCallback` hook to make React remember these methods and improve performance